File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {

        int[] countArr = new int[n+1];

        for(int i=0; i<=n; i++){
            int num = i;
            while(num != 0){
                if((num & 1) == 1){
                    countArr[i]++;
                }
                num = num >> 1;
            }
        }

        return countArr;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:

        arr = []
        
        for num in range(n+1):
            count = 0
            while num > 0:
                if num & 1 == 1:
                    count += 1
                num = num >> 1
            arr.append(count)

        return arr
